src/Raven.Server/RavenServer.cs

```diff
@@ -1560,15 +1703,15 @@ private bool TryAuthorize(RavenConfiguration configuration, Stream stream, TcpCo
             }
 
             var certificate = (X509Certificate2)sslStream.RemoteCertificate;
-            var auth = AuthenticateConnectionCertificate(certificate);
+            var auth = AuthenticateConnectionCertificate(certificate, remoteAddress);
 
             switch (auth.Status)
             {
                 case AuthenticationStatus.Expired:
-                    msg = "The provided client certificate " + certificate.FriendlyName + " is expired on " + certificate.NotAfter;
+                    msg = "The provided client certificate " + certificate?.FriendlyName + " is expired on " + certificate?.NotAfter;
```

Comment:
```suggestion
                    msg = "The provided client certificate " + (certificate?.FriendlyName  ?? "<Missing cert>")+ " is expired on " + certificate?.NotAfter;
```

As well as elsewhere around here

Reply:
and how can this be null?

Reply:
It cannot be null here. 
It can be null in general but then we'll geth auth.Status = NoCertificateProvided which will make us fall to the default case in the switch.

I believe I put the '?' there because of a compiler warning about the casting a few lines above.
var certificate = (X509Certificate2)sslStream.RemoteCertificate;

Reply:
That shouldn't cause it to warn about null, I think.
Let's drop this `?`